### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by toddmedema**
refactor and fix checkout
switch from pre-commit to pre-push
fix audio controls
live-calculate timer duration based on up to date settings to allow for changing adventurers setting mid-combat

### CODE REVIEW

File: app/components/views/Checkout.tsx [hunk 1]
```diff
@@ -77,37 +78,50 @@ export default class Checkout extends React.Component<CheckoutProps, {}> {
...
+            {this.props.checkout.processing && <div className="centralMessage">Processing payment, one moment...</div>}
+          </form>
+        </div>
+      </Card>
+    );
+  }
+}
+
+
+class CheckoutThankYou extends React.Component<CheckoutProps, {}> {
```
Comment (smartin015) [inline app/components/views/Checkout.tsx]: Probably best if this were just a function (like [these](https://github.com/ExpeditionRPG/expedition-app/blob/master/app/components/views/quest/cardtemplates/combat/Combat.tsx#L59))

File: app/components/views/quest/cardtemplates/combat/Actions.tsx [hunk 2]
```diff
@@ -19,7 +19,8 @@ import {getStore} from '../../../../../Store'
 const cheerio: any = require('cheerio');
 
 // Number of adventurers (such as for damage)
-function numLocalAndMultiplayerAdventurers(settings: SettingsType, rp: MultiplayerState): number {
+// TODO test
```
Comment (smartin015) [inline app/components/views/quest/cardtemplates/combat/Actions.tsx]: Move this to the test file, add stubs (here & below), and remove the comment on L21 (I don't think it adds value)

File: app/components/views/quest/cardtemplates/combat/Actions.tsx [hunk 3]
```diff
@@ -19,7 +19,8 @@ import {getStore} from '../../../../../Store'
...
+export function _numLocalAndMultiplayerAdventurers(settings: SettingsType, rp: MultiplayerState): number {
```
Comment (smartin015) [inline app/components/views/quest/cardtemplates/combat/Actions.tsx]: `export` implies "used elsewhere", leading underscore implies local. Which one's correct?

File: app/components/views/quest/cardtemplates/combat/Actions.tsx [hunk 4]
```diff
@@ -37,7 +38,8 @@ function numLocalAndMultiplayerAdventurers(settings: SettingsType, rp: Multiplay
 }
 
 // Number of fingers
-function numLocalAndMultiplayerers(settings: SettingsType, rp: MultiplayerState): number {
+// TODO test
+export function _numLocalAndMultiplayerers(settings: SettingsType, rp?: MultiplayerState): number {
```
Comment (smartin015) [inline app/components/views/quest/cardtemplates/combat/Actions.tsx]: `Multiplayerers`?! What is this thing?

File: app/components/views/quest/cardtemplates/combat/Combat.tsx [hunk 5]
```diff
@@ -440,7 +440,7 @@ function renderTimerCard(props: CombatProps): JSX.Element {
       tertiaryText={instruction}
       icon={enemyClass}
       numPlayers={(props.settings.multitouch && props.settings.numPlayers > 1) ? props.numAliveAdventurers : 1}
-      roundTimeTotalMillis={props.combat.roundTimeMillis}
```
Comment (smartin015) [inline app/components/views/quest/cardtemplates/combat/Combat.tsx]: Do we not need `roundTimeMillis` in CombatState any more?

File: app/components/views/quest/cardtemplates/combat/Actions.tsx (hunk 4 above)
Comment (toddmedema) [reply app/components/views/quest/cardtemplates/combat/Actions.tsx]: lol, done

File: app/components/views/quest/cardtemplates/combat/Combat.tsx (hunk 5 above)
Comment (toddmedema) [reply app/components/views/quest/cardtemplates/combat/Combat.tsx]: Nope, it's a calculated parameter. If we wanted to get really fancy here, we might do something like a selector as the official way to handle derived data - but that doesn't seem worth getting into right now. https://redux.js.org/recipes/computing-derived-data

File: app/components/views/quest/cardtemplates/combat/Actions.tsx (hunk 3 above)
Comment (toddmedema) [reply app/components/views/quest/cardtemplates/combat/Actions.tsx]: Only exported for tests. Open to other ways of handling that :)

File: app/components/views/quest/cardtemplates/combat/Actions.tsx (hunk 2 above)
Comment (toddmedema) [reply app/components/views/quest/cardtemplates/combat/Actions.tsx]: Done

File: app/components/views/Checkout.tsx (hunk 1 above)
Comment (toddmedema) [reply app/components/views/Checkout.tsx]: Done

File: app/components/views/quest/cardtemplates/combat/Combat.tsx (hunk 5 above)
Comment (smartin015) [reply app/components/views/quest/cardtemplates/combat/Combat.tsx]: If it's calculated, let's remove it from `CombatState` defined in cardtemplates/combat/Types.tsx.

File: app/components/views/quest/cardtemplates/combat/Actions.tsx (hunk 3 above)
Comment (smartin015) [reply app/components/views/quest/cardtemplates/combat/Actions.tsx]: As discussed, let's privatize these and instead test them indirectly through the interface.

File: app/components/views/quest/cardtemplates/combat/Combat.tsx (hunk 5 above)
Comment (toddmedema) [reply app/components/views/quest/cardtemplates/combat/Combat.tsx]: It is removed from there already

**Pull request merged**